### PR TITLE
Merge DFL kernel latest patches

### PR DESCRIPTION
**Pull request by Muralix**

### CODE REVIEW

**Review by rweight — DISMISSED** [review]
I reviewed the changes by doing a comparison against the linux-dfl implementation and looking at the backport-specific changes. The changes look good to me.

File: include/linux/minmax.h [hunk 1]
```diff
@@ -0,0 +1,14 @@
+/* SPDX-License-Identifier: GPL-2.0-only */
+#ifndef __BACKPORT_MINMAX_H
+#define __BACKPORT_MINMAX_H
+
+#include <linux/version.h>
+
+
+#if LINUX_VERSION_CODE < KERNEL_VERSION(5, 10, 0)
```
Comment (rchriste1) [inline include/linux/minmax.h]: The linux/minmax.h file is included i CentOS8.5 and the following RockyLinux 8.6 and 8.7. So the criteria should be:

#if LINUX_VERSION_CODE < KERNEL_VERSION(5, 10, 0) && RHEL_RELEASE_CODE <= 0x804

**Review by rchriste1 — APPROVED** [review]
I have run a simple test on N5010 on Rocky Linux 9.1 and compiled OK on relevant OS's.

**Comment by rweight** [comment]
I made a couple of small tweaks and then pushed the code from my working branch to both the intel-test/fpga-ofs-dev-5.15-lts and intel/fpga-ofs-dev-5.15-lts branches. The changes I made are:

(1) Spelling fixes in a commit message
(2) I moved the backport fix for the UART driver to right after the UART driver changes
(3) I squashed the minmax.h fix with the patch that created it
(4) I dropped the Kconfig fix because it pertains to the firmware loader, but we use the FPGA image load driver for the backport driver (Kconfig files don't get used for the backport, but in the interest of not creating unnecessary confusion, I'm trying to keep it correct)

**Pull request closed without merging**